modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/io.scala

```diff
@@ -492,8 +491,47 @@ object io {
     azureStorage: Option[BlobStorageClients.AzureStorage]
   )
   object BlobStorageClients {
-    case class AzureStorage(storageAccountNames: List[String])
 
+    case class AzureStorage(accounts: List[AzureStorage.Account])
+    object AzureStorage {
+
+      final case class Account(name: String, auth: Option[Account.Auth])
+      object Account {
+
+        sealed trait Auth
+        object Auth {
+          final case object CredentialsChain extends Auth
+          final case class SasToken(value: String) extends Auth
+        }
+      }
+    }
+
+    implicit val sasTokenDecoder: Codec[AzureStorage.Account.Auth.SasToken] =
+      deriveConfiguredCodec
+
+    implicit val accountAuthDecoder: Decoder[AzureStorage.Account.Auth] =
+      Decoder.instance { cursor =>
+        val typeCur = cursor.downField("type")
+        typeCur.as[String].map(_.toLowerCase) match {
+          case Right("chain") =>
```

Comment:
IMO "default" would be a better type name to put in the config file.  Meaning Azure's _default_ credentials chain.